---
layout: post
title: FROGER Day 02
date: '2020-03-06'
categories: FROGER
tags: [mox, methylation, bismark]
---

## Annotating Mcap Proteins

The genome version at http://cyanophora.rutgers.edu/montipora/ has proteins without annotations. Will give those a whirl.

![pic](http://gannet.fish.washington.edu/seashell/snaps/Genome_analysis_of_the_rice_coral_Montipora_capitata_2020-03-06_08-09-04.png)

First I had to clean up hardwrap

```
%%bash
awk '!/^>/ { printf "%s", $0; n = "\n" }
/^>/ { print n $0; n = "" }
END { printf "%s", n }
' /Volumes/web/tmp/Mcap.protein.fasta  > /Volumes/web/tmp/Mcap.protein.fa
```

Then blastp

```
#!/bin/bash
## Job Name
#SBATCH --job-name=blastp
## Allocation Definition
#SBATCH --account=coenv
#SBATCH --partition=coenv
## Resources
## Nodes
#SBATCH --nodes=1
## Walltime (days-hours:minutes:seconds format)
#SBATCH --time=1-00:00:00
## Memory per node
#SBATCH --mem=120G
##turn on e-mail notification
#SBATCH --mail-type=ALL
#SBATCH --mail-user=sr320@uw.edu
## Specify the working directory for this job
#SBATCH --chdir=/gscratch/scrubbed/sr320/030620-Mcap-annotation


# Load Python Mox module for Python module availability

module load intel-python3_2017




source /gscratch/srlab/programs/scripts/paths.sh



/gscratch/srlab/programs/ncbi-blast-2.6.0+/bin/blastp \
-query /gscratch/srlab/sr320/data/froger/Mcap/Mcap.protein.fa \
-db /gscratch/srlab/blastdbs/UniProtKB_20190109/uniprot_sprot.fasta \
-num_threads 28 \
-outfmt 6 \
-max_target_seqs 1 \
-evalue 1e-05 \
-out /gscratch/scrubbed/sr320/030620-Mcap-annotation/Mcap_blastp_sp.tab
```
