---
layout: post
title: What's a Weka going to do?
date: '2022-01-22'
categories: oyster
tags: RNA-seq, ML
---

Here I want to examine how Machine Learning might compare with a our conventional gene expression analysis. The data set includes both male and female oysters exposed to OA conditions (and controls). Gonad tissue. Sam ran data through bowtie/stringtie for comparison. Complete sample details are below.

|Sample.ID|OldSample.ID|Treatment|Sex|TreatmentN|Parent.ID|
|---------|------------|---------|---|----------|---------|
|12M      |S12M        |Exposed  |M  |3         |EM05     |
|13M      |S13M        |Control  |M  |1         |CM04     |
|16F      |S16F        |Control  |F  |2         |CF05     |
|19F      |S19F        |Control  |F  |2         |CF08     |
|22F      |S22F        |Exposed  |F  |4         |EF02     |
|23M      |S23M        |Exposed  |M  |3         |EM04     |
|29F      |S29F        |Exposed  |F  |4         |EF07     |
|31M      |S31M        |Exposed  |M  |3         |EM06     |
|35F      |S35F        |Exposed  |F  |4         |EF08     |
|36F      |S36F        |Exposed  |F  |4         |EF05     |
|39F      |S39F        |Control  |F  |2         |CF06     |
|3F       |S3F         |Exposed  |F  |4         |EF06     |
|41F      |S41F        |Exposed  |F  |4         |EF03     |
|44F      |S44F        |Control  |F  |2         |CF03     |
|48M      |S48M        |Exposed  |M  |3         |EM03     |
|50F      |S50F        |Exposed  |F  |4         |EF01     |
|52F      |S52F        |Control  |F  |2         |CF07     |
|53F      |S53F        |Control  |F  |2         |CF02     |
|54F      |S54F        |Control  |F  |2         |CF01     |
|59M      |S59M        |Exposed  |M  |3         |EM01     |
|64M      |S64M        |Control  |M  |1         |CM05     |
|6M       |S6M         |Control  |M  |1         |CM02     |
|76F      |S76F        |Control  |F  |2         |CF04     |
|77F      |S77F        |Exposed  |F  |4         |EF04     |
|7M       |S7M         |Control  |M  |1         |CM01     |
|9M       |S9M         |Exposed  |M  |3         |EM02     |

---

Sam provided me with a [table of FPKM values](https://d.pr/f/7sitpg) pulled from Ballgown. There is information for about 39k genes.
In short I believe Sam got no DEGs with record to treatment, but as might be expect many differences based on sex.

![table-pic](http://gannet.fish.washington.edu/seashell/snaps/gene_fpkm_2022-01-22_13-43-01.png)

I opened the csv file up in the Weka Explorer.

![exp](http://gannet.fish.washington.edu/seashell/snaps/Weka_Explorer_2022-01-22_13-47-39.png)

Then ran Classifier using `SMO`, on Treatment (thus ignoring sex).

![class](http://gannet.fish.washington.edu/seashell/snaps/Weka_Explorer_2022-01-22_13-49-00.png)

I went on to Select Attributes

![sa](http://gannet.fish.washington.edu/seashell/snaps/Weka_Explorer_2022-01-22_13-51-52.png)


Based on this I found [499 genes](https://d.pr/n/4S4atL) contributing ranging in values from 0.2388 to 0.7743.

My goal now is to rerun this pipeline in Weka just using data from these 499 genes to see if the predictability improves. Will be doing some inelegant data munging with bbedit, excel, and Rstudio.

Step 1) simply read in Sam's fpkm table with 39k genes.

```
fpkm <- read.csv(file = "../analyses/gene_fpkm.csv")
```

Step 2) generate list of genes I can insert into a dplyr::select function. (skip this part if adverse to cringing)

Paste into excel, copy gene column, paste back into bbedit:    
Text::remove line breaks    
Find::replace `-` with `.` (due to Weka formatting?)  
Find::replace ` ` with `,`     
Paste in dplyr::select code chunk
